### ISSUE TITLE
get_player_control() not returning input from mobile clients

### ISSUE THREAD

**Issue by GenshinMT** [minetest/minetest#10438]
## Problem

Minetest on mobile should return or simulate player control inputs when receiving inputs from function **get_player_control()**
at the moment it currently isn't (returns everything false). which doesn't make any sense. and renders this function useless for those who are playing on the mobile version of the client.

## Solutions

A way/alternative method to get control inputs from mobile client users.

**Comment by jamesalexanderstevenson** [comment]
Works for me.

**Comment by SmallJoker** [comment]
If `get_player_control` isn't working in your use-case, please check whether you can see other player's model animations (sneak, run, dig). Those also depend on `get_player_control`.

**Comment by GenshinMT** [comment]
> If `get_player_control` isn't working in your use-case, please check whether you can see other player's model animations (sneak, run, dig). Those also depend on `get_player_control`.

At client version 5.0.1, **get_player_control()** isn't returning control input feedback for mobile users (returns false whenever a mobile user is pressing punch or place). I checked for this behaviour using this method:
`minetest.register_on_placenode(function(pos, newnode, placer, oldnode, itemstack, pointed_thing)
 local right_click = placer:get_player_control().RMB

  --Is player right clicking while placing a node?
  if right_click == true then
    minetest.chat_send_all(placer:get_player_name().." is right clicking while placing nodes")
  elseif right_click == false then
    minetest.chat_send_all(placer:get_player_name().." is not right clicking while placing nodes")
  end

end)`
perhaps is a issue with previous versions based on feedback that I gathered from others.

**Comment by GenshinMT** [comment]
Ran some test based on what SmallJoker suggested, looks like its due to timing issues.

### Evidence

[When mobile user is Digging while holding key](https://cdn.discordapp.com/attachments/306656385754464258/760215472364060692/Screenshot_20200928-150343_Minetest.jpg)

[When PC user is placing nodes while holding key](https://cdn.discordapp.com/attachments/306656385754464258/760215676936781844/Screenshot_at_2020-09-28_14-44-17.png)

[When mobile user is placing nodes while unable to hold down place key](https://cdn.discordapp.com/attachments/306656385754464258/760215538373885962/Screenshot_20200928-144142_Minetest.jpg)

**Comment by wsor4035** [comment]
can confirm this issue

**Comment by sfan5** [comment]
If you can confirm this issue can you be more specific which part? Did you use the RMB example code?

**Comment by SmallJoker** [comment]
**About the original issue**

`ObjectRef:get_player_control` depends on the packet `TOSERVER_PLAYERPOS`, which is:
1. ... sent periodically, every `m_recommended_send_interval = 0.1f` seconds and
2. ... unreliable (i.e. transmission success is not checked)

~~Hence, the controls getter function is not guaranteed to return matching key presses in callbacks triggered by interactions. It seems to be that this issue is a feature request either for documentation or a kind of change request for the existing behaviour.~~ INVALID. See relevant code below.
https://github.com/luanti-org/luanti/blob/65d498f78e609feff37fcdfb7aa5a321e5733536/src/client/client.cpp#L1102-L1106

---

@emperorgenshin That is a duplicate of #13029

**Comment by sfan5** [comment]
> the controls getter function is not guaranteed to return matching key presses in callbacks triggered by interactions

This is not true IIRC. The `TOSERVER_INTERACT` packet contains a copy of the player control state to avoid precisely this problem.
